tools/src/bin/output-file.rs

```diff
@@ -320,17 +427,184 @@ fn main() {
+                // bother with strip_prefix and rebuilding paths, although if we
+                // were creating URLs to services, we would want that.
+                let meta_ini_path = path.replacen("/tests/", "/meta/", 1);
+                let meta_url = format!(
+                    // The .ini extension gets appended onto the path, retaining
+                    // the existing file extension.
+                    r#"/{}/source/{}.ini"#,
+                    tree_name,
+                    meta_ini_path,
+                );
+
+                // Track whether we emitted something with the meta URL.
+                let mut linked_meta = false;
+
+                if wpt_info.disabling_conditions.len() > 0 {
+                    linked_meta = true;
+                    list_nodes.push(F::Seq(vec![
+                        F::S("<li>"),
+                        F::Indent(vec![
+                            F::T(format!(
+                                r#"This test has a <a href="{}">WPT meta file</a> that disables it given conditions:"#,
+                                meta_url
+                            )),
+                            F::S("<ul>"),
+                            F::Indent(wpt_info.disabling_conditions.iter().map(|(cond, bug)| {
+                                F::T(format!(
+                                    r#"<li><span class="test-skip-info">{}</span>&nbsp; : <a href="https://bugzilla.mozilla.org/buglist.cgi?quicksearch={}">{}</a></li>"#,
+                                    // The condition text can embed newlines at the end.
+                                    cond.trim(),
+                                    bug,
+                                    bug,
+                                ))
+                            }).collect()),
+                            F::S("</ul>"),
+                        ]),
+                        F::S("</li>"),
+                    ]));
+                }
+
+                if wpt_info.disabled_subtests_count > 0 {
+                    linked_meta = true;
+                    list_nodes.push(F::T(format!(
+                        r#"<li>This test has a <a href="{}">WPT meta file</a> that expects {} subtest issues."#,
+                        meta_url,
+                        wpt_info.disabled_subtests_count,
+                    )));
+                }
+
+                // If we didn't emit bullets above that have the link, then emit a vague message
+                // with the link.
+                if !linked_meta {
+                    list_nodes.push(F::T(format!(
+                        r#"<li>This test has a <a href="{}">WPT meta file</a> for some reason."#,
+                        meta_url,
+                    )));
+                }
+            }
+
+            list_nodes.push(F::T(format!(
+                "<li>This test ran {} times in the preceding 7 days with an average run time of {:.2} secs.</li>",
```

Comment:
I noticed "inf secs" being reported on some pages, so maybe make the text after "preceding 7 days" conditional on the number of unskipped runs being > 0.